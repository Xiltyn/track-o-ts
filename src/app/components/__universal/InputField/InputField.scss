@import "../../../style/core/variables";
@import "../../../style/utils/functions";
@import "../../../style/utils/mixins";

.input-container {
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    padding-bottom: 24px;

    label {
        font-size: 16px;
        font-weight: 500;
        line-height: 20px;
        padding-bottom: 8px;

        transition: all .3s ease-in-out;
    }

    input {
        background: transparent;
        border: 1px solid $GREY_DIM;
        border-radius: 2px;
        padding: 8px 12px;

        font-size: 16px;
        line-height: 16px;
        color: $GREY_DIM;

        transition: all .3s ease-in-out;

        &::placeholder {
            color: $GREY_PALE;
        }

        &:hover,
        &:active {
            border-color: $PRIMARY;
            color: $PRIMARY;
        }
    }

    &.active {
        label {
            color: $PRIMARY;
        }

        input {
            border-color: $PRIMARY;
            color: $PRIMARY;
        }
    }
}
