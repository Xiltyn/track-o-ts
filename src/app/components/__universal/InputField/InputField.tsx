import * as React from 'react';
import { Field } from "redux-form";

import './InputField.scss';

export namespace InputField {
    export interface Props {
        name: string,
        type: 'text'|'password'|'number'|'file',
        label?: string,
        className?: string,
        placeholder?: string,
    }

    export interface State {
        isActive: boolean,
    }
}

export class InputField extends React.Component<InputField.Props, InputField.State> {
    static defaultProps:Pick<InputField.Props, 'type'> = {
        type: 'text',
    };

    state:Pick<InputField.State, 'isActive'> = {
        isActive: false,
    };

    render() {
        const {
            name,
            type,
            label,
            className,
            placeholder,
        } = this.props;

        const {
            isActive,
        } = this.state;

        return(
            <div
                className={ `input-container ${ isActive ? 'active' : '' }` }
                onMouseEnter={ () => this.setState({ isActive: true }) }
                onMouseLeave={ () => this.setState({ isActive: false }) }>
                {
                    label && <label htmlFor={ name }>
                        { label }
                    </label>
                }
                <Field
                    name={ name }
                    className={ className ? className : '' }
                    type={ type }
                    placeholder={ placeholder && placeholder }
                    component="input"/>
            </div>
        )
    }
}
