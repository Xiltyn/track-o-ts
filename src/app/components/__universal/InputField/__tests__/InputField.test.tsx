import * as React from 'react';
import { InputField } from '../InputField';
import { shallow, ShallowWrapper } from "enzyme";

describe('<|:===== <InputField /> =====:|>', () => {
    const wrapper:ShallowWrapper<InputField.Props> = shallow(
        <InputField
            name='fake_name'/>
    );

    it('renders component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
