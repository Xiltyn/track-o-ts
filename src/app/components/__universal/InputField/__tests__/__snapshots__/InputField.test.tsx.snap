// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<|:===== <InputField /> =====:|> renders component 1`] = `
<div
  className="input-container "
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Field
    className=""
    component="input"
    name="fake_name"
    type="text"
  />
</div>
`;
